### PR TITLE
[native] Table writer 1: Pass LocationHandle to HiveInsertTableHandle

### PR DESCRIPTION
**Pull request by gggrace14**
Enable the ser/de of protocol::HiveInsertTableHandle.

Extract LocationHandle and HiveColumnHandles from
protocol::HiveOutputTableHandle and protocol::HiveInsertTableHandle,
and add them to Velox connector::HiveInsertTableHandle.
Pass them down to the plan TableWriteNode.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 1]
```diff
@@ -112,6 +100,46 @@ std::shared_ptr<connector::ColumnHandle> toColumnHandle(
...
+    protocol::TableType tableType) {
+  switch (tableType) {
+    case protocol::TableType::NEW:
+      return connector::hive::LocationHandle::TableType::kNew;
+    case protocol::TableType::EXISTING:
+      return connector::hive::LocationHandle::TableType::kExisting;
+    case protocol::TableType::TEMPORARY:
+      return connector::hive::LocationHandle::TableType::kTemporary;
+    default:
+      throw std::invalid_argument("Unknown table type");
```
Comment (amitkdutta) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:113]: Might be better to throw velox exception from here.

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 2]
```diff
@@ -112,6 +100,46 @@ std::shared_ptr<connector::ColumnHandle> toColumnHandle(
...
+      return connector::hive::LocationHandle::WriteMode::
+          kStageAndMoveToTargetDirectory;
+    case protocol::WriteMode::DIRECT_TO_TARGET_NEW_DIRECTORY:
+      return connector::hive::LocationHandle::WriteMode::
+          kDirectToTargetNewDirectory;
+    case protocol::WriteMode::DIRECT_TO_TARGET_EXISTING_DIRECTORY:
+      return connector::hive::LocationHandle::WriteMode::
+          kDirectToTargetExistingDirectory;
+    default:
+      throw std::invalid_argument("Unknown write mode");
```
Comment (amitkdutta) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:130]: ditto

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 3]
```diff
@@ -1086,7 +1114,7 @@ core::PlanNodePtr VeloxQueryPlanConverter::toVeloxQueryPlan(
           node->source)) {
     std::optional<core::JoinType> joinType = std::nullopt;
     if (equal(node->predicate, semiJoin->semiJoinOutput)) {
-      joinType = core::JoinType::kLeftSemi;
+      joinType = core::JoinType::kLeftSemiFilter;
```
Comment (amitkdutta) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1117]: Is this change related here?

File: presto-native-execution/CMakeLists.txt [hunk 4]
```diff
@@ -165,6 +165,7 @@ include_directories(.)
 include_directories(velox)
 include_directories(velox/velox/external/xxhash)
 include_directories(${VELOX_ROOT})
+include_directories(${CMAKE_BINARY_DIR})
```
Comment (gggrace14) [inline presto-native-execution/CMakeLists.txt:168]: This is to fix the proto path issue triggered by this PR.

'velox/velox/dwio/dwrf/proto/dwrf_proto.pb.h' file not found

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp (hunk 3 above)
Comment (gggrace14) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1117]: This is to pick up Masha's change, per ask

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp (hunk 1 above)
Comment (gggrace14) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:113]: Got feedback on this in https://github.com/prestodb/presto/pull/18377, and address all occurrences in this file there.

Corresponding code copy of this PR has already been merged into fbcode repo. This PR is supposed to be just porting to GitHub, so better to keep the code identical of the fbcode copy. It's just Merge was blocked by the protobuf path issue as well as velox_hive_connector OBJECT issue as in https://github.com/facebookincubator/velox/pull/3094.

**Comment by mbasmanova** [comment]
@gggrace14 CI join is still failing. Do you know why?

**Comment by gggrace14** [comment]
> Reference in n

Ch

**Comment by gggrace14** [comment]
> @gggrace14 CI join is still failing. Do you know why?

Checking it

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp (hunk 1 above)
Comment (spershin) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:113]: @gggrace14 and everyone.
We do not want to port fbcode changes to GH. We want to avoid any changes in fbcode.
We will get this through, though.

@gggrace14 can you please create and follow up a small issue on changing this exception to Velox exception, please? To be consistent.

**Review by spershin — APPROVED** [review]
@gggrace14 Let's get this in.

**Comment by gggrace14** [comment]
Tracking the troubleshooting process for dependency issues

1. [Error] Could not make proto path relative: velox/dwio/dwrf/proto/dwrf_proto.proto: No such file or directory
[Fix] https://github.com/facebookincubator/velox/pull/3062

2. [Error] fatal error: 'velox/velox/dwio/dwrf/proto/orc_proto.pb.h' file not found
#include "velox/velox/dwio/dwrf/proto/orc_proto.pb.h"
[Fix] Add to [presto-native-execution/CMakeLists.txt](https://github.com/prestodb/presto/pull/18490/files#diff-f4770996c945761a121b4351ae31adecae46b00a0767c8c4cb04112731984795)
```
include_directories(${CMAKE_BINARY_DIR})
```

3.
